src/IO/ITKImageInputFileFormat.cxx

```diff
@@ -140,76 +139,113 @@ read_from_file(const std::string& filename) const
-            origin,
-            voxel_size);
-
-    // copy data
-    VoxelsOnCartesianGrid<float>::full_iterator stir_iter = image_ptr->begin_all();
-    typedef itk::ImageRegionConstIterator< ITKImageSingle > IteratorType;
-    IteratorType it (itk_image, itk_image->GetLargestPossibleRegion() );
-    for ( it.GoToBegin(); !it.IsAtEnd(); ++it, ++stir_iter  ) {
-        *stir_iter = static_cast<float>(it.Get());
-    }
-
-    return image_ptr;
+  ITKImageSingle::Pointer itk_image = itk_image_orig->Clone();
+  STIRImageSingle* image_ptr = construct_empty_stir_image_and_orient_itk_image
+    <ITKImageSingle, STIRImageSingle, STIRImageSingleConcrete>(itk_image);
+
+  // copy data
+  VoxelsOnCartesianGrid<float>::full_iterator stir_iter = image_ptr->begin_all();
+  typedef itk::ImageRegionConstIterator<ITKImageSingle> IteratorType;
+  IteratorType it (itk_image, itk_image->GetLargestPossibleRegion());
+  for (it.GoToBegin(); !it.IsAtEnd(); ++it, ++stir_iter)
+  {
+    *stir_iter = static_cast<float>(it.Get());
+  }
+  return image_ptr;
 }
 
 // Actual conversion function
 STIRImageMulti*
 convert_ITK_to_STIR(const ITKImageMulti::Pointer itk_image_orig)
 {
-    ITKImageMulti::Pointer itk_image;
-    CartesianCoordinate3D<float> voxel_size;
-    BasicCoordinate<3,int> min_indices, max_indices;
-    CartesianCoordinate3D<float> origin;
-
-    // orientate the ITK image
-    orient_ITK_image<ITKImageMulti>(
-                itk_image, voxel_size, min_indices, max_indices, origin, itk_image_orig);
-
-    // create STIR image
-    STIRImageMulti* image_ptr =
-        new STIRImageMulti
-            (IndexRange<3>(min_indices, max_indices),
-             origin,
-             voxel_size);
-
-    // copy data
-    STIRImageMulti::full_iterator stir_iter = image_ptr->begin_all();
-    typedef itk::ImageRegionConstIterator< ITKImageMulti > IteratorType;
-    IteratorType it (itk_image, itk_image->GetLargestPossibleRegion() );
-    for ( it.GoToBegin(); !it.IsAtEnd(); ++it, ++stir_iter) {
-        itk::Point<double,3U> itk_coord;
-
-        itk_coord[0] = -double(it.Get()[0]);
-        itk_coord[1] = -double(it.Get()[1]);
-        itk_coord[2] =  double(it.Get()[2]);
-
-        *stir_iter =
-                ITK_coordinates_to_STIR(
-                    itk_coord,
-                    voxel_size,
-                    min_indices,
-                    true);
-    }
-    return image_ptr;
+  ITKImageMulti::Pointer itk_image = itk_image_orig->Clone();
+  STIRImageMulti* image_ptr = construct_empty_stir_image_and_orient_itk_image
+    <ITKImageMulti, STIRImageMulti, STIRImageMulti>(itk_image);
+
+  // copy data
+  STIRImageMulti::full_iterator stir_iter = image_ptr->begin_all();
+  typedef itk::ImageRegionConstIterator<ITKImageMulti> IteratorType;
+  IteratorType it (itk_image, itk_image->GetLargestPossibleRegion() );
+  for ( it.GoToBegin(); !it.IsAtEnd(); ++it, ++stir_iter) {
+      itk::Point<double,3U> itk_coord;
+
+      itk_coord[0] = -double(it.Get()[0]);
```

Comment:
why the minus signs here?

Reply:
@rijobro I assume this can go now, I'll remove it. Best you double check on your data.